### PR TITLE
BSD auth functions

### PR DESCRIPTION
**Pull request by dantecatalfamo**

### CODE REVIEW

**Comment by Vexu** [comment]
This pr has been a draft for over a month with no activity, is it blocking on something?

**Comment by dantecatalfamo** [comment]
I've just been quite busy recently, sorry. I'll try to get it ready to merge soon!

File: lib/std/c/openbsd.zig [hunk 1]
```diff
@@ -45,6 +45,52 @@ pub extern "c" fn unveil(path: ?[*:0]const u8, permissions: ?[*:0]const u8) c_in
...
+pub extern "c" fn auth_clroption(as: *auth_session_t, option: [*:0]const u8) void;
+pub extern "c" fn auth_clroptions(as: *auth_session_t) void;
+pub extern "c" fn auth_setenv(as: *auth_session_t) void;
+pub extern "c" fn auth_getvalue(as: *auth_session_t, what: [*:0]const u8) ?[*:0]const u8;
+pub extern "c" fn auth_verify(as: ?*auth_session_t, style: ?[*:0]const u8, name: ?[*:0]const u8, ...) ?*auth_session_t;
+pub extern "c" fn auth_call(as: *auth_session_t, path: [*:0]const u8, ...) c_int;
+pub extern "c" fn auth_challenge(as: *auth_session_t) [*:0]const u8;
+pub extern "c" fn auth_check_expire(as: *auth_session_t) i64;
+pub extern "c" fn auth_check_change(as: *auth_session_t) i64;
+// TODO auth_getpwd requires passwd struct
```
Comment (Vexu) [inline lib/std/c/openbsd.zig:90]: `struct passwd;` in Zig is `const passwd = opaque{};`.

Comment (dantecatalfamo) [reply lib/std/c/openbsd.zig]: I just finished filling out the passwd struct and passwd-related functions. Should I replace the `extern struct` with an `opaque`?

Comment (Vexu) [reply lib/std/c/openbsd.zig]: It looks to be a part of the public API so it should be fine.

**Pull request merged**